### PR TITLE
Add changes from rector rule RemoveNullPropertyInitializationRector

### PR DESCRIPTION
**Pull request by Harfusha**
https://github.com/cakephp/cakephp/issues/17756

### CODE REVIEW

File: src/Validation/ValidationRule.php [hunk 1]
```diff
@@ -40,7 +40,7 @@ class ValidationRule
      *
      * @var callable|string|null
      */
-    protected $_on = null;
+    protected $_on;
```
Comment (ADmad) [inline src/Validation/ValidationRule.php:43]: This doesn't make sense to me. Removing the null initialization will cause an `not initialized error` if the property is accessed without explicitly setting a value.

Comment (Harfusha) [reply src/Validation/ValidationRule.php:43]: Both variables are same: https://stackoverflow.com/questions/14097258/is-a-property-default-value-of-null-the-same-as-no-default-value

Comment (dereuromark) [reply src/Validation/ValidationRule.php:43]: Not if you set a type for it afaik. Which here seems to not be the case I guess

Comment (Harfusha) [reply src/Validation/ValidationRule.php:43]: Isn't this typed property? public int $id;

Comment (ADmad) [reply src/Validation/ValidationRule.php:43]: Ah yes, this is an untyped property since `callable` can't be used.

**Pull request merged**